### PR TITLE
Add snmp_settings to eos_designs

### LINKED ISSUE #683: SNMP location abstraction
**Issue Type**

- [ ] Template enhancement
- [x] Role enhancement
- [x] Documentation enhancement (avd.sh)

**Is your feature request related to a problem? Please describe.**

If we are setting SNMP location on devices, it will not be possible to automate this with a single knob using `cli_config_gen`, unless we define per host this variable.

**Describe the solution you'd like**

It would be great if we can abstract this for `eos_l3ls_evpn` such as 

`snmp_location: {{ fabric_name }} {{ dc_name }} {{ pod_name}} {{ inventory_hostname }}`

or better if we can leverage the use of `rack: RackA` and make this include geo location / which rack the DUT is on, which could be placed under `node/node_group`, similar to what we have with server definition.

`snmp_location: {{ fabric_name }} {{ dc_name }} {{ pod_name}} {{ rack }} {{ inventory_hostname }}`

In the end we would populate this per switch using the information already provided in group_vars, and rack + hostname making it unique per switch.

**Describe alternatives you've considered**

Configlet with script maybe.

**Additional context**

Of course this would be optional knob.

### PR DESCRIPTION
**Pull request by melkypie**
## Change Summary

Allows people to define their contact info and whether or not to use a pre-formatted location (see #683 for more info)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [X] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

Fixes #683

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

Adds a new data model (snmp_settings) to eos_designs
Also adds the ability to label specific devices to racks
```yaml
snmp_settings:
  contact: < contact_info >
  location: < boolean | default -> false >
```

```yaml
  node_groups:

    # node_group_1, will result in stand-alone leaf.
    < node_group_1 >:

      # All variables defined under `defaults` will be inherited by the node group, if not specifically set inside it.

      # Arista platform family. | Required
      platform: < Arista Platform Family >

      # Rack that the switch is located in (only used in snmp_settings location) | Optional
      rack: < rack_name >
```

## Checklist:

- [X] My code has been rebased from devel before I start
- [X] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [X] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/doc/l3ls-evpn/common-settings.md [hunk 1]
```diff
@@ -87,6 +87,11 @@ mac_address_table:
 platform_speed_groups:
   < platform >:
     < speed >: [ < speed_group >, < speed_group > ]
+
+# Set SNMP settings | Optional
+snmp_settings:
+  contact: < contact_info >
+  location: < boolean | default -> false >
```
Comment (titom73) [inline ansible_collections/arista/avd/roles/eos_designs/doc/l3ls-evpn/common-settings.md]: Could we add a description to explain role of location as boolean? From the code I understand it enables logic to build location string.

**Review by titom73 — APPROVED** [review]
LGTM! Only a minor comment to add.

Thanks for your contribution !

**Review by titom73 — APPROVED** [review]
LGTM!

**Review by carlbuchmann — APPROVED** [review]
LGTM, thanks for this PR!

**Pull request merged**